src/mono/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.Mono.cs

```diff
@@ -1898,20 +1897,19 @@ private static bool IsValidGetMethodType(Type type)
         public static MethodInfo GetMethod(Type type, MethodInfo method)
         {
             if (!IsValidGetMethodType(type))
-                throw new ArgumentException("type is not TypeBuilder but " + type.GetType(), nameof(type));
+                throw new ArgumentException("type must contain a TypeBuilder as a generic argument.", nameof(type));
```

Comment:
NIT: a bit inconsistency with a period, looks they don't put a period on not capitalized exception messages

Reply:
shouldn't this be a in a resx?

Reply:
Yeah, not sure why there are lots of string literals used in this file instead of using resources

Reply:
> Yeah, not sure why there are lots of string literals used in this file instead of using resources

Historical reasons. When this code was initially imported from mono/mono it made sense to keep the implementations similar for each of patching them in sync.  The mono/mono code didn't use string resources.
At this point it's better to use the same string resources are CoreCLR, if possible.

Reply:
as an aside, it's my understanding that any replacement strings that aren't just numbers need quotes. This is because if we ever localize, the localizers can't make the sentence grammatical without knowing what the replacement string is.

so `"type is not TypeBuilder but " + type.GetType() + " instead."` would not be allowable but something like this would be : `"type is not TypeBuilder but '" + type.GetType() + "' instead."`

`'type' is not TypeBuilder.` would be OK, but not as helpful because it doesn't include the type.

Reply:
> shouldn't this be a in a resx?

Yes, according to the bug you logged last year https://github.com/dotnet/runtime/issues/34056 😄

Reply:
I've fixed the exception messages I touched to use the same resources as CoreCLR. I left the ones I didn't touch since they don't correspond directly to existing CoreCLR exception messages. Those can get fixed with #34056.